internal/supervisor/installation.go

```diff
@@ -322,6 +342,64 @@ func (s *InstallationSupervisor) createInstallation(installation *model.Installa
 	return model.InstallationStateCreationNoCompatibleClusters
 }
 
+// prioritizeLowerUtilizedClusters attempts filter the given cluster list and
+// order it by lowest resource usage first. This should be considered best
+// effort.
+// Note the following:
+//   - This check is done without locking to avoid creating additional
+//     congestion.
+//   - Resource usage ordering is done by taking an average of CPU + memory
+//     percentages.
+//   - The returned list will generally be in order of lowest-to-highest
+//     resource usage, but the only guarantee is that the first cluster in the
+//     list will be the lowest at the time it was checked.
+//   - When scheduling an installation, all of the standard scheduling checks
+//     should be performed again under cluster lock.
+func (s *InstallationSupervisor) prioritizeLowerUtilizedClusters(clusters []*model.Cluster, installation *model.Installation, instanceID string, logger log.FieldLogger) []*model.Cluster {
+	lowestResourcePercent := 10000
+	var filteredPrioritizedClusters []*model.Cluster
+
+	for _, cluster := range clusters {
+		if !s.installationCanBeScheduledOnCluster(cluster, installation, logger) {
+			continue
+		}
+
+		clusterResources, err := s.provisioner.GetClusterResources(cluster, true)
```

Comment:
GetClusterResources() gives you an accurate used & total CPU/Memory for a cluster but what happens when there are 10 clusters and 100s of nodes? Would you be scanning every cluster on every install?

Even though they say it is less accurate, would it be better to use metrics-server? (not in this PR but for the future)

https://github.com/kubernetes-sigs/metrics-server

Reply:
Yes, you are absolutely correct that this resource calculation is relatively expensive and will take longer with every cluster and node added.

Having a hard toggle via the server startup flag will allow us to use this behavior only when really needed and we fully intend to go with a more-scalable approach when time permits. I will absolutely take a fresh look at the metrics server option to see how much effort it would take to implement.

Thanks for the comment!

Reply:
Yeah, this definitely could lead to longer scheduling times so we should keep an eye on it and improve when we have other metric options ready.